www/js/services.js

```diff
@@ -48,12 +48,13 @@ angular.module('pvta.services', ['ngResource'])
 
 
 .factory('StopList', function(){
-  var stopsList = [];
-  var pushToList = function(stop){
-    stopsList.push(stop);
-  };
+  var stopsList = []; 
+
   var pushEntireList = function(list){
-    stopsList = stopsList.concat(list);
+    list = _.uniq(list, true, 'Name');
```

Comment:
A second parameter of `true` only works if the list is already sorted by the attribute you're uniquifying by, but that doesn't look like it's the case here ( it was here but it was removed? ).